packages/connect/src/device/Device.ts

```diff
@@ -567,6 +573,10 @@ export class Device extends TypedEmitter<DeviceEvents> {
         const { message } = await this.getCommands().typedCall('GetFeatures', 'Features', {});
         this._updateFeatures(message);
 
+        if (this.authenticityChecks.firmwareHash === null) {
+            this.authenticityChecks.firmwareHash = await this.checkFirmwareHash();
```

Comment:
now it's boilerplate - **if** we decide this is the way to go, I'll refactor the common parts of this function in subsequent PR